// @virtusize/design-system/dist/lib/Modal/Modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _compositionApi = require("@vue/composition-api");

var _Modal = _interopRequireDefault(require("./Modal.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Modal = (0, _compositionApi.createComponent)({
  name: "Modal",
  props: {
    zIndex: Number,
    onClick: Object,
    componentClass: {
      type: Array,
      default: function _default() {
        return [];
      }
    }
  },
  setup: function setup(props, context) {
    var onClick = function onClick(event) {
      context.emit("click", event);
    };

    return function () {
      var _context$slots$defaul, _context$slots;

      return (0, _compositionApi.createElement)("div", {
        "class": [_Modal.default.root].concat(props.componentClass),
        "style": {
          zIndex: props.zIndex
        },
        "on": {
          "click": onClick
        }
      }, [(_context$slots$defaul = (_context$slots = context.slots).default) === null || _context$slots$defaul === void 0 ? void 0 : _context$slots$defaul.call(_context$slots)]);
    };
  }
});
var _default2 = Modal;
exports.default = _default2;
