// @virtusize/design-system/dist/lib/TabBar/TabBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _compositionApi = require("@vue/composition-api");

var _TextView = _interopRequireDefault(require("../TextView"));

var _TabBar = _interopRequireDefault(require("./TabBar.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Used to indicate the current active screen.
 * @param {array} componentClass - Add custom classes on the component.
 * @param {array} tabs - Set the array of tabs. A tab must be of format { label: "Label", value: "value" }.
 * @param {string} value - Select the current Tab.
 */
var TabBar = (0, _compositionApi.createComponent)({
  name: "TabBar",
  props: {
    componentClass: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    onChange: Object,
    tabs: {
      type: Array,
      required: true
    },
    value: String
  },
  setup: function setup(props, context) {
    var onClick = function onClick(value) {
      context.emit("change", value);
    };

    return function () {
      return (0, _compositionApi.createElement)("div", {
        "class": [_TabBar.default.root].concat(props.componentClass)
      }, [props.tabs.map(function (_ref) {
        var label = _ref.label,
            value = _ref.value;
        return (0, _compositionApi.createElement)(_TextView.default, {
          "attrs": {
            "size": "small",
            "weight": value === props.value ? "bold" : "medium",
            "componentClass": [_TabBar.default.item, value === props.value ? _TabBar.default.selected : ""]
          }
        }, [(0, _compositionApi.createElement)("a", {
          "on": {
            "click": function click() {
              return onClick(value);
            }
          }
        }, [label])]);
      })]);
    };
  }
});
var _default2 = TabBar;
exports.default = _default2;
