// @virtusize/design-system/dist/lib/TextField/TextField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _compositionApi = require("@vue/composition-api");

var _TextView = _interopRequireDefault(require("../TextView"));

var _TextField = _interopRequireDefault(require("./TextField.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TextField = (0, _compositionApi.createComponent)({
  name: "TextField",
  props: {
    componentClass: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    error: Boolean,
    fluid: Boolean,
    message: String,
    onChange: Object,
    onClick: Object,
    onKeyup: Object,
    onBlur: Object,
    onFocus: Object,
    placeholder: String,
    type: String,
    value: String
  },
  setup: function setup(props, context) {
    var isFocused = (0, _compositionApi.ref)(false);

    var handleEvent = function handleEvent(event) {
      event.stopPropagation();
      if (event.type === "focus") isFocused.value = true;
      if (event.type === "blur") isFocused.value = false;

      if (event.type === "click") {
        context.refs.input.focus();
      }

      context.emit(event.type, event.target.value, event);
    };

    return function () {
      return (0, _compositionApi.createElement)("div", {
        "class": [_TextField.default.root, isFocused.value ? _TextField.default.focus : "", props.error ? _TextField.default.error : "", props.fluid ? _TextField.default.fluid : ""].concat(props.componentClass),
        "on": {
          "click": handleEvent
        }
      }, [context.slots.leading && (0, _compositionApi.createElement)("span", {
        "class": _TextField.default.leading
      }, [context.slots.leading()]), (0, _compositionApi.createElement)("input", {
        "attrs": {
          "type": props.type,
          "placeholder": props.placeholder
        },
        "class": _TextField.default.input,
        "on": {
          "focus": handleEvent,
          "blur": handleEvent,
          "change": handleEvent,
          "keyup": handleEvent
        },
        "domProps": {
          "value": props.value
        },
        "ref": "input"
      }), context.slots.trailing && (0, _compositionApi.createElement)("span", {
        "class": _TextField.default.trailing
      }, [context.slots.trailing()]), props.message && (0, _compositionApi.createElement)(_TextView.default, {
        "attrs": {
          "tag": "p",
          "componentClass": [_TextField.default.message],
          "size": "small"
        }
      }, [props.message])]);
    };
  }
});
var _default2 = TextField;
exports.default = _default2;
