// @virtusize/design-system/dist/lib/VisualPicker/VisualPicker.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _compositionApi = require("@vue/composition-api");

var _TextView = _interopRequireDefault(require("../TextView"));

var _VisualPicker = _interopRequireDefault(require("./VisualPicker.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Select an item in a more visual, appealing way.
 * @param {array} componentClass - Add custom classes on the component
 * @param {boolean} fluid - Set VisualPicker to use all available horizontal space.
 * @param {onClick} onClick - Callback function to invoke when VisualPicker is clicked
 * @param {boolean} selected - Set VisualPicker to be selected
 */
var VisualPicker = (0, _compositionApi.createComponent)({
  name: "VisualPicker",
  props: {
    componentClass: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    fluid: Boolean,
    onClick: Object,
    selected: Boolean
  },
  setup: function setup(props, context) {
    var onClick = function onClick(event) {
      context.emit("click", event);
    };

    return function () {
      var _context$slots$icon, _context$slots, _context$slots$defaul, _context$slots2;

      return (0, _compositionApi.createElement)("div", {
        "class": [_VisualPicker.default.root, props.fluid ? _VisualPicker.default.fluid : "", props.selected ? _VisualPicker.default.selected : ""].concat(props.componentClass),
        "on": {
          "click": onClick
        }
      }, [(0, _compositionApi.createElement)("div", {
        "class": _VisualPicker.default.icon
      }, [(_context$slots$icon = (_context$slots = context.slots).icon) === null || _context$slots$icon === void 0 ? void 0 : _context$slots$icon.call(_context$slots)]), (0, _compositionApi.createElement)(_TextView.default, {
        "attrs": {
          "size": "small",
          "weight": "bold",
          "align": "right",
          "block": true
        }
      }, [(_context$slots$defaul = (_context$slots2 = context.slots).default) === null || _context$slots$defaul === void 0 ? void 0 : _context$slots$defaul.call(_context$slots2)])]);
    };
  }
});
var _default2 = VisualPicker;
exports.default = _default2;
