// @virtusize/design-system/dist/lib/ProgressBar/ProgressBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _compositionApi = require("@vue/composition-api");

var _ProgressBar = _interopRequireDefault(require("./ProgressBar.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Show the length of a process.
 *
 * @param {string} color - Specify the color of the progress bar. Any valid CSS color (hex, rgb, rgba, linear-gradient, etc) is accepted.
 * @param {string} backgroundColor - Specify the color of the progress bar background. Any valid CSS color (hex, rgb, rgba, linear-gradient, etc) is accepted.
 * @param {number} value - Set the progress between 1-100 (in percentage, whole integers only).
 * @param {array} componentClass - Add custom classes on the component.
 */
var ProgressBar = (0, _compositionApi.createComponent)({
  name: "ProgressBar",
  props: {
    color: String,
    backgroundColor: String,
    value: {
      type: Number,
      required: true,
      validator: function validator(prop) {
        return prop > -1 && prop <= 100;
      }
    },
    componentClass: {
      type: Array,
      default: function _default() {
        return [];
      }
    }
  },
  setup: function setup(props) {
    return function () {
      return (0, _compositionApi.createElement)("div", {
        "class": [_ProgressBar.default.root].concat(props.componentClass),
        "style": {
          background: props.color
        }
      }, [(0, _compositionApi.createElement)("div", {
        "class": [_ProgressBar.default.container]
      }, [(0, _compositionApi.createElement)("div", {
        "class": _ProgressBar.default.progress,
        "style": {
          background: props.backgroundColor,
          transform: "translateX(".concat(props.value - 100, "%)"),
          borderRadius: "".concat(props.value === 100 ? 0 : 16, "px")
        }
      })])]);
    };
  }
});
var _default2 = ProgressBar;
exports.default = _default2;
