// @virtusize/design-system/dist/lib/Strip/StripItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _compositionApi = require("@vue/composition-api");

var _symbols = require("./symbols");

var _StripItem = _interopRequireDefault(require("./StripItem.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * An individual item in Strip.
 * @param {array} componentClass - Add custom classes on the component.
 * @param {string} value - This item's value. Required prop.
 */
var StripItem = (0, _compositionApi.createComponent)({
  name: "StripItem",
  props: {
    componentClass: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    value: {
      type: String,
      required: true
    }
  },
  setup: function setup(props, context) {
    var state = (0, _compositionApi.inject)(_symbols.VALUE);
    var selected = (0, _compositionApi.inject)(_symbols.SELECTED_VALUE);
    return function () {
      var _context$slots$defaul, _context$slots;

      return (0, _compositionApi.createElement)("div", {
        "ref": "root",
        "class": [_StripItem.default.root, props.value === selected ? _StripItem.default.selected : "", props.value !== state.value ? _StripItem.default.hidden : ""].concat(props.componentClass),
        "attrs": {
          "data-value": props.value
        }
      }, [(_context$slots$defaul = (_context$slots = context.slots).default) === null || _context$slots$defaul === void 0 ? void 0 : _context$slots$defaul.call(_context$slots)]);
    };
  }
});
var _default2 = StripItem;
exports.default = _default2;
