// @virtusize/design-system/dist/lib/Avatar/Avatar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _compositionApi = require("@vue/composition-api");

var _Avatar = _interopRequireDefault(require("./Avatar.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * A representation of something (e.g. a user, an icon)
 * @param {string} borderColor - Set the border color.
 * @param {string} borderStyle - Set the border style. Accepted values are dashed, none, or solid. Defaults to solid. Ignored when borderColor is set.
 * @param {string} borderWidth - Set the border thickness. Accepted values are thin, thick. Defaults to thick.
 * @param {array} componentClass - Add custom classes on the component.
 * @param {boolean} elevated - Show a shadow behind the component.
 * @param {string} size - Set the size of the component. Accepted values are small and large. Defaults to small.
 * @param {boolean} inverse - set the inner background color to black instead of white
 */
var Avatar = (0, _compositionApi.createComponent)({
  name: "Avatar",
  props: {
    borderColor: String,
    borderStyle: {
      type: String,
      default: function _default() {
        return "solid";
      }
    },
    borderWidth: {
      type: String,
      default: function _default() {
        return "thin";
      }
    },
    componentClass: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    elevated: Boolean,
    size: {
      type: String,
      default: function _default() {
        return "small";
      }
    },
    inverse: Boolean
  },
  setup: function setup(props, context) {
    var styleObject = (0, _compositionApi.computed)(function () {
      if (props.borderStyle !== "dashed") return {
        borderColor: props.borderColor
      };
      if (props.borderColor) return {
        background: props.borderColor,
        borderColor: props.inverse ? "#000" : "#fff"
      };
      return {
        background: "linear-gradient(315deg, #16c6b9 0%, #19f0b0 100%)",
        borderColor: props.inverse ? "#000" : "#fff"
      };
    });
    return function () {
      var _context$slots$defaul, _context$slots;

      return (0, _compositionApi.createElement)("div", {
        "class": [_Avatar.default.root, _Avatar.default[props.size], _Avatar.default[props.borderStyle], _Avatar.default[props.borderWidth], props.elevated ? _Avatar.default.elevated : ""].concat(props.componentClass),
        "style": styleObject.value
      }, [(0, _compositionApi.createElement)("div", {
        "class": _Avatar.default.innerWrapper,
        "style": {
          background: props.inverse ? "#000" : "#fff"
        }
      }, [(_context$slots$defaul = (_context$slots = context.slots).default) === null || _context$slots$defaul === void 0 ? void 0 : _context$slots$defaul.call(_context$slots)])]);
    };
  }
});
var _default2 = Avatar;
exports.default = _default2;
