// @virtusize/design-system/dist/lib/Sheet/Sheet.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _compositionApi = require("@vue/composition-api");

var _Button = _interopRequireDefault(require("../Button"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _Modal = _interopRequireDefault(require("../Modal"));

var _Sheet = _interopRequireDefault(require("./Sheet.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Display activities in the current context.
 * @param {array} componentClass - Add custom classes on the component.
 * @param {boolean} open - Open Sheet.
 * @param {function} onClose - Callback function to invoke when Sheet is closed
 * @param {function} onEnter - Callback function to invoke when Sheet is opened
 * @param {function} onExit - Callback function to invoke before Sheet is closed
 * @param {number} zIndex - Set Sheet's z-index.
 */
var Sheet = (0, _compositionApi.createComponent)({
  name: "Sheet",
  props: {
    componentClass: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    maxHeight: String,
    open: {
      type: Boolean,
      required: true
    },
    onClose: Object,
    onEnter: Object,
    onExit: Object,
    zIndex: Number
  },
  setup: function setup(props, context) {
    var isSheetOpen = (0, _compositionApi.ref)(false);
    var modal;
    var sheet;

    var invokeTransitionHook = function invokeTransitionHook(event) {
      event.stopPropagation();
      isSheetOpen.value = props.open;
      context.emit(props.open ? "enter" : "exit");

      if (!props.open) {
        sheet.removeEventListener("transitionend", invokeTransitionHook);
      }
    };

    (0, _compositionApi.watch)(function () {
      return props.open;
    }, function _callee(open) {
      var _ref;

      return regeneratorRuntime.async(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              if (!open) {
                _context.next = 9;
                break;
              }

              isSheetOpen.value = open;
              _context.next = 4;
              return regeneratorRuntime.awrap(context.root.$nextTick());

            case 4:
              modal = (_ref = context.refs.modal) === null || _ref === void 0 ? void 0 : _ref.$el;
              sheet = context.refs.sheet;

              if (modal && sheet) {
                modal.addEventListener("transitionend", invokeTransitionHook);
                setTimeout(function () {
                  modal.classList.add(_Sheet.default.open);
                  sheet.classList.add(_Sheet.default.open);
                }, 100);
              }

              _context.next = 10;
              break;

            case 9:
              if (modal && sheet) {
                modal.classList.remove(_Sheet.default.open);
                sheet.classList.remove(_Sheet.default.open);
              }

            case 10:
            case "end":
              return _context.stop();
          }
        }
      });
    });
    return function () {
      var _context$slots$defaul, _context$slots;

      return isSheetOpen.value && (0, _compositionApi.createElement)(_Modal.default, {
        "attrs": {
          "componentClass": [_Sheet.default.modal].concat(props.componentClass),
          "zIndex": props.zIndex
        },
        "on": {
          "click": function click() {
            context.emit("close");
          }
        },
        "ref": "modal"
      }, [(0, _compositionApi.createElement)("div", {
        "class": [_Sheet.default.sheet],
        "on": {
          "click": function click(event) {
            return event.stopPropagation();
          }
        },
        "style": {
          maxHeight: props.maxHeight
        },
        "ref": "sheet"
      }, [(0, _compositionApi.createElement)("div", {
        "class": _Sheet.default.controls
      }, [(0, _compositionApi.createElement)(_Button.default, {
        "attrs": {
          "round": true,
          "noBorder": true,
          "small": true
        },
        "on": {
          "click": function click() {
            return context.emit("close");
          }
        }
      }, [(0, _compositionApi.createElement)(_Icon.default, {
        "attrs": {
          "componentClass": [_Sheet.default.close],
          "name": "angleDown",
          "size": "small"
        }
      })])]), (_context$slots$defaul = (_context$slots = context.slots).default) === null || _context$slots$defaul === void 0 ? void 0 : _context$slots$defaul.call(_context$slots)])]);
    };
  }
});
var _default2 = Sheet;
exports.default = _default2;
